homeassistant/components/mysensors.py

```diff
@@ -56,25 +60,47 @@ def setup(hass, config):  # pylint: disable=too-many-locals
     is_metric = hass.config.units.is_metric
     persistence = config[DOMAIN].get(CONF_PERSISTENCE, True)
 
-    def setup_gateway(device, persistence_file, baud_rate, tcp_port):
+    def setup_gateway(device, persistence_file, baud_rate, tcp_port, in_prefix,
+                      out_prefix):
         """Return gateway after setup of the gateway."""
-        try:
-            socket.inet_aton(device)
-            # valid ip address
-            gateway = mysensors.TCPGateway(
-                device, event_callback=None, persistence=persistence,
-                persistence_file=persistence_file, protocol_version=version,
-                port=tcp_port)
-        except OSError:
-            # invalid ip address
-            gateway = mysensors.SerialGateway(
-                device, event_callback=None, persistence=persistence,
-                persistence_file=persistence_file, protocol_version=version,
-                baud=baud_rate)
+        # pylint: disable=too-many-arguments
+        if device in 'mqtt':
+            if not setup_component(hass, 'mqtt', config):
+                return
+            mqtt = get_component('mqtt')
```

Comment:
I think @turbokongen means that I should extract `mqtt` into a constant, ie
```py
MQTT_COMPONENT = 'mqtt'
```
I think that doesn't hurt right?